Remove support for the hspace and vspace on table

Ahmad-S792 opened the pull request:
#### 6676a3759ed36734def1b35f6d40d9c160b6fd7f
<pre>
Remove support for the hspace and vspace on table

Remove support for the hspace and vspace attributes on table

<a href="https://bugs.webkit.org/show_bug.cgi?id=120133">https://bugs.webkit.org/show_bug.cgi?id=120133</a>

Reviewed by Ryosuke Niwa.

Partial Merge - Since Webkit does not use &quot;User Counter&quot; - <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=195601

It will align Webkit with Blink, Gecko and Web-spec.

* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/tables/table-vspace-hspace-expected.txt: Updated Expectations to &apos;PASS&apos;.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/tables/table-vspace-hspace-s-expected.txt: Updated Expectations to &apos;PASS&apos;.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/unmapped-attributes-expected.txt: Updated Expectations to &apos;PASS&apos;.
* LayoutTests/platform/glib/tables/mozilla/bugs/bug1261-expected.txt: Rebased RenderTable.
* LayoutTests/platform/gtk/tables/mozilla/bugs/bug1261-expected.png: Updated based on new expectations.
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1261-expected.txt: Rebased RenderTable.
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1261-expected.png: Updated based on new expectations.
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1261-expected.txt: Rebased RenderTable.
* Source/WebCore/html/HTMLTableElement.cpp
(HTMLTableElement::collectPresentationalHintsForAttribute): Removed &apos;hspace&apos; and &apos;vspace&apos; attributes on table to match with all other browsers.

Remove support for the hspace and vspace on table

Canonical link: <a href="https://commits.webkit.org/253788@main">https://commits.webkit.org/253788@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ea272cb1ad3b996f72a0c0b24cdfcc6650b0361f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87023 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31109 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/17867 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/95949 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149603 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91000 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29474 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/25757 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79156 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91053 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92639 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/23796 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/73846 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23782 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/78784 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79057 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66791 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27206 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/12895 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27148 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/13909 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2676 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28831 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/36760 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28772 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33186 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on LayoutTests/imported/w3c/web-platform-tests/html/rendering/unmapped-attributes-expected.txt:
```diff
@@ -2,14 +2,14 @@
-FAIL <table vspace> should not be mapped to style marginTop in BackCompat mode assert_equals: expected "0px" but got "200px"
-FAIL <table vspace> should not be mapped to style marginBottom in CSS1Compat mode assert_equals: expected "0px" but got "200px"
-FAIL <table vspace> should not be mapped to style marginBottom in BackCompat mode assert_equals: expected "0px" but got "200px"
+PASS <table hspace> should not be mapped to style marginLeft in CSS1Compat mode assert_equals: expected "0px" but got "200px"
```
Looks like you've got extra stuff at the end of each one of these lines.

rniwa replied:
e.g. see https://ews-build.s3-us-west-2.amazonaws.com/macOS-BigSur-Release-WK2-Tests-EWS/22872081-13895/imported/w3c/web-platform-tests/html/rendering/unmapped-attributes-pretty-diff.html

Ahmad-S792 replied:
Fixed and pushed updated commit. 👍

Commit:
Remove support for the hspace and vspace on table

Remove support for the hspace and vspace attributes on table

https://bugs.webkit.org/show_bug.cgi?id=120133

Reviewed by Ryosuke Niwa.

Partial Merge - Since Webkit does not use "User Counter" - https://src.chromium.org/viewvc/blink?view=revision&revision=195601

It will align Webkit with Blink, Gecko and Web-spec.

* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/tables/table-vspace-hspace-expected.txt: Updated Expectations to 'PASS'.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/tables/table-vspace-hspace-s-expected.txt: Updated Expectations to 'PASS'.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/unmapped-attributes-expected.txt: Updated Expectations to 'PASS'.
* LayoutTests/platform/glib/tables/mozilla/bugs/bug1261-expected.txt: Rebased RenderTable.
* LayoutTests/platform/gtk/tables/mozilla/bugs/bug1261-expected.png: Updated based on new expectations.
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1261-expected.txt: Rebased RenderTable.
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1261-expected.png: Updated based on new expectations.
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1261-expected.txt: Rebased RenderTable.
* Source/WebCore/html/HTMLTableElement.cpp
(HTMLTableElement::collectPresentationalHintsForAttribute): Removed 'hspace' and 'vspace' attributes on table to match with all other browsers.

Remove support for the hspace and vspace on table

Canonical link: https://commits.webkit.org/253788@main
